executor/src/witgen/fixed_evaluator.rs

```diff
@@ -22,19 +21,24 @@ impl<'a, T: FieldElement> FixedEvaluator<'a, T> {
 }
 
 impl<'a, T: FieldElement> SymbolicVariables<T> for FixedEvaluator<'a, T> {
```

Comment:
The FixedEvaluator is only used in the SortedWitnessMachine where we try to detect the structure of the machine.

Should we remove it?

Reply:
Yes, it's used to detect `NOTLAST $ [ A' - A ] in [ POSITIVE ]`, where `NOTLAST` and `POSITIVE` have the expected structure.

I think the benefit of using the evaluator is that it works for any expression, e.g. `NOTLAST = 1 - LAST`. But I think we could also use `SymbolicWitnessEvaluator` (which implements the same logic, but delegates the witness access to a member implementing `WitnessColumnEvaluator`) with a `WitnessColumnEvaluator` that panics when called.

But seems outside the scope of this PR.